Framework/Core/include/Framework/ASoA.h

```diff
@@ -1015,20 +1015,15 @@ constexpr bool is_binding_compatible_v()
   return are_bindings_compatible_v<T>(originals_pack_t<B>{});
 }
 
+//! Helper to check if a type T is an iterator
 template <typename T>
-using is_soa_iterator_t = typename framework::is_base_of_template<RowViewCore, T>;
-
-template <typename T>
-constexpr bool is_soa_iterator_v()
-{
-  return is_soa_iterator_t<T>::value || framework::is_specialization_v<T, RowViewCore>;
-}
+inline constexpr bool is_soa_iterator_v = framework::is_base_of_template_v<RowViewCore, T>;
```

Comment:
I doubt that this is equivalent to the previous code.

Reply:
I have my doubts about it as well, but why would:

```c++
framework::is_specialization_v<T, RowViewCore>;
```

and

```C++
framework::is_base_of_template_v<RowViewCore, T>
```

differ?

Reply:
ok, I have aligned the two versions. As discussed, I am not sure it actually matters, since hyperloop seems fine with it.